pkg/proxy/config/api_test.go

```diff
@@ -174,24 +165,16 @@ func TestNewEndpointsSourceApi_UpdatesAndMultipleEndpoints(t *testing.T) {
 
 	ch := make(chan EndpointsUpdate)
 
-	cache.NewReflector(lw, &api.Endpoints{}, NewEndpointsStore(nil, ch), 30*time.Second).Run()
-
-	got, ok := <-ch
-	if !ok {
-		t.Errorf("Unable to read from channel when expected")
-	}
-	expected := EndpointsUpdate{Op: SET, Endpoints: []api.Endpoints{}}
-	if !apiequality.Semantic.DeepEqual(expected, got) {
-		t.Errorf("Expected %#v; Got %#v", expected, got)
-	}
+	endpointsController := NewEndpointsController(lw, 30*time.Second, ch)
+	go endpointsController.Run(wait.NeverStop)
```

Comment:
I think by #1 here you mean "Make sure that the List has fully completed before we do anything with the state", and a big +1 on that...  I think we've seen that bug in a few controllers.  But ... did this bug also exist previously (if not, can someone point me to magic where it was avoided?)

(Also in general, if anyone has a good doc on Informers vs Reflectors vs direct-watching.  I prefer direct-watching, but I guess that means I have all sort of edge cases in my code.)